quarto/Group_10_Analysis.qmd

```diff
@@ -0,0 +1,188 @@
+---
+title: "Analysis of IMDB Rating with GLM"
+format: html
+editor: visual
+---
+
+# Introduction {#sec-intro}
+
+Given the properties of films, predict whether the rating will be greater than 7 or not.
+
+```{r}
+#| echo: true
+#| warning: false
+
+library(tidyverse)
+library(gt)
+
+df <- read_csv("../data/dataset10.csv")
+
+df |> 
+  slice_head(n=5) |>
+  gt() |>
+  cols_label(
+    film_id = html("Film ID"),
+    year = html("Year"),
+    length = html("Length"),
+    budget = html("Budget"),
+    votes = html("Votes"),
+    genre = html("Genre"),
+    rating = html("Rating")
+  )
+```
+
+Description:
+
+-   Film ID: The unique identifier for the film
+-   Year: Year of release of the film in cinemas
+-   Length: Duration (in minutes)
+-   budget: Budget for the films production (in \$1000000s)
+-   votes: Number of positive votes received by viewers
+-   genre: Genre of the film
+-   rating: IMDB rating from 0-10
+
+# Data Wrangling
+
+## Missing value
+
+```{r}
+#| echo: true
+#| warning: false
+
+# Calculate the number of NA
+# Dataset10 is replaced by we want to use
+
+na_sum <- colSums(is.na(df))
+
+par(ask = FALSE, 
+    mfrow = c(1, 1), 
+    mar = c(7, 6, 2, 2))
+
+barplot(na_sum, 
+        main = "Missing Values Count", 
+        ylab = "Count",                
+        col = "skyblue",              
+        names.arg = colnames(df), 
+        ylim = c(0, max(na_sum) + 200),
+        las=2,
+        adj=1) 
+
+text(x = 1:length(na_sum)+0.2, y = na_sum + 2, labels = na_sum, pos = 3, cex = 0.8, col = "red")
+```
+
+```{r}
```

Comment:
for each type of movie, select them individually, and then calculate their average rating? then we can know what kind of movies have higher rating.

Reply:
See how many unique values in the categorical variables